### PR TITLE
chore: Refactor of `ProfileContextMenu` usages and related components

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

This PR simplifies and organizes the code of the components below as as step toward eliminating dependency on `Utils.getContactDetails` and other backend-dependent singleton methods:

- `ProfileContextMenu` - minor API improvements, `Storybook` page made fully operable
- `member_model` - `isCurrentUser` role exposed
- Unused and outdated `ContactRequestPanel` removed
- `UserListPanel` made pure UI component, no dependencies on stores and global utils calling backend, fully operable Storybook page added
- `MembersTabPanel` - handling profile context menu refactored
- `MessageView`  - handling profile context menu refactored
- `UserImage` API changed to make the component backend-independent (calls to `Utils.colorIdForPubkey` and `Utils.getColorHashAsJson` removed), usages adapted

Closes: #16717 (some not covered areas related to `Utils.getContactDetails` will be covered in follow-up PRs)

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

Screenshot from 2024-11-08 14-50-19
Screenshot from 2024-11-08 14-50-01

### Impact on end user

- no changes observable from user perspective

### How to test

- All interactions with profile context menu from all locations (community members list, community settings, settings/contacts, message view)

Tick **one**:
- [x] Low risk: 2 devs MUST perform testing as specified above and attach their results as comments to this PR **before** merging.
- [ ] High risk: QA team MUST perform additional testing in the specified affected areas **before** merging.

### CODE REVIEW

**Comment by caybro** [comment]
Ouch, I did some of the stuff here as well: https://github.com/status-im/status-desktop/pull/16714

**Comment by virginiabalducci** [comment]
Tests passed

File: ui/imports/utils/Utils.qml [hunk 1]
```diff
@@ -700,6 +700,28 @@ QtObject {
...
+        if (isMe)
+            return Constants.profileType.self
+        if (isBridgedAccount)
+            return Constants.profileType.bridged
+
+        return isBlocked ? Constants.profileType.blocked
+                         : Constants.profileType.regular
+    }
+
+    function getContactType(contactRequestState, isContact) {
```
Comment (caybro) [inline ui/imports/utils/Utils.qml:713]: This is weird (and maybe duplicate); we can infer the `isContact` already if `contactRequestState === Constants.ContactRequestState.Mutual`

**Review by caybro — APPROVED** [review]
LGTM, nice work!

**Pull request merged**